Sources/PackageGraph/PackageGraphLoader.swift

```diff
@@ -25,6 +25,27 @@ struct UnusedDependencyDiagnostic: DiagnosticData {
     public let dependencyName: String
 }
 
+public struct DuplicateDependencyDiagnostic: DiagnosticData, Swift.Error {
+    public static var id = DiagnosticID(
+        type: DuplicateDependencyDiagnostic.self,
+        name: "org.swift.diags.duplicate-dependency",
+        defaultBehavior: .warning,
+        description: {
+            $0 <<< "dependency" <<< { "'\($0.dependencyName)'" } <<< "already exists;"
+            $0 <<< "merging requirements" <<< { $0.versionText }
```

Comment:
Merging seems misleading here. We should say something like:

found multiple declaration for dependency 'Foo'; using requirement  `.upToNextMajor(from: 2.0.0)`.